tests/common/gcp_type/firewall_rule_test.py

```diff
@@ -55,6 +55,286 @@ def test_ips_subset_of_ips(self, ips, ips_range, expected):
                 'firewall_rule_source_tags': None,
                 'firewall_rule_direction': 'ingress',
                 'firewall_rule_network': 'n1',
+                'firewall_rule_priority': 'NaN',
+                'firewall_rule_allowed': json.dumps(
+                    [{'IPProtocol': 'tcp', 'ports': ['22']}]),
+            },
+        ),
+        (
+            {
+                'firewall_rule_source_ranges': json.dumps(['1.1.1.1']),
+                'firewall_rule_direction': 'ingress',
+                'firewall_rule_network': 'n1',
+                'firewall_rule_priority': '-1',
+                'firewall_rule_allowed': json.dumps(
+                    [{'IPProtocol': 'tcp', 'ports': ['22']}]),
+            },
+        ),
+        (
+            {
+                'firewall_rule_source_ranges': json.dumps(['1.1.1.1']),
+                'firewall_rule_direction': 'ingress',
+                'firewall_rule_network': 'n1',
+                'firewall_rule_priority': '1000000000',
+                'firewall_rule_allowed': json.dumps(
+                    [{'IPProtocol': 'tcp', 'ports': ['22']}]),
+            },
+        ),
+    ])
+    def test_validate_priority_error(self, rule_dict):
+        rule = firewall_rule.FirewallRule(**rule_dict)
+        with self.assertRaises(firewall_rule.InvalidFirewallRuleError):
+            rule._validate_priority()
+
+    @parameterized.parameterized.expand([
+        (
```

Comment:
nit: Just a general observation, it would be nice to have a comment inside of each tuple that explains what the test condition / expectation is. Like '\# ingress rule has no source ranges'

Reply:
Done.